src/main/java/com/brandwatch/internship/fetchandstoretweets/entities/Query.java

```diff
@@ -0,0 +1,37 @@
+package com.brandwatch.internship.fetchandstoretweets.entities;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Query {
+
+    private static AtomicLong counter = new AtomicLong();
+    private final long id;
+    private final String searchString;
+
+    public Query(String searchString) {
+        this.id = counter.incrementAndGet();
```

Comment:
Won't this just give any query `id = 1`?

You can't give the query the id from inside the application, it has to come from the database. So for now it should just be part of the constructor params.

Reply:
Yeah, this caused a problem that of course I could not notice before the service branch. The problem was that when I use findAll() on the query repository, it creates new queries from the documents it fetches. This lead to giving new ids to the new queries. I solved this problem in the service branch and made it like what you said, id in constructor.

Reply:
but the ids will not be the same cause the counter is incremented every time.

Reply:
Ok if you solve it in your next branch then it's fine.

>but the ids will not be the same cuz the counter is incremented every time.

Yeah but for every new Query it's a new `AtomicLong` that starts from 0 right?

Reply:
Ohhh it's `static`. Ignore me.